contracts/account-abstraction/Account.sol

```diff
@@ -0,0 +1,402 @@
+import "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+import "@openzeppelin/contracts/interfaces/IERC1271.sol";
+import "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+
+import {BaseAccount as BaseERC4337Account, IEntryPoint, UserOperation} from "./base-account-abstraction/core/BaseAccount.sol";
+
+import {ParaAccount} from "./ParaAccount.sol";
+
+error NotAuthorized();
+error InvalidInput();
+error AccountLocked();
+error ExceedsMaxLockTime();
+error UntrustedImplementation();
+error OwnershipCycle();
+
+/**
+ * @title A smart contract account owned by a single ERC721 token
+ */
+contract Account is
+    IERC165,
+    IERC1271,
+    IERC6551Account,
+    IERC721Receiver,
+    IERC1155Receiver,
+    UUPSUpgradeable,
+    BaseERC4337Account
+{
+    using ECDSA for bytes32;
+
+    /// @dev ERC-4337 entry point address
+    address public immutable _entryPoint;
+
+    /// @dev timestamp at which this account will be unlocked
+    uint256 public lockedUntil;
+
+    /// @dev mapping from owner => selector => implementation
+    mapping(address => mapping(bytes4 => address)) public overrides;
+
+    /// @dev mapping from owner => caller => has permissions
+    mapping(address => mapping(address => bool)) public permissions;
+
+    event OverrideUpdated(
+        address owner,
+        bytes4 selector,
+        address implementation
+    );
+
+    event PermissionUpdated(address owner, address caller, bool hasPermission);
+
+    event LockUpdated(uint256 lockedUntil);
+
+    /// @dev reverts if caller is not the owner of the account
+    modifier onlyOwner() {
+        if (msg.sender != owner()) revert NotAuthorized();
+        _;
+    }
+
+    /// @dev reverts if caller is not authorized to execute on this account
+    modifier onlyAuthorized() {
+        if (!isAuthorized(msg.sender)) revert NotAuthorized();
+        _;
+    }
+
+    /// @dev reverts if this account is currently locked
+    modifier onlyUnlocked() {
+        if (isLocked()) revert AccountLocked();
+        _;
+    }
+
+    constructor(address entryPoint_) {
+        if (entryPoint_ == address(0)) revert InvalidInput();
+
+        _entryPoint = entryPoint_;
+    }
+
+    /// @dev allows eth transfers by default, but allows account owner to override
+    receive() external payable {
+        // _handleOverride();
+    }
+
+    /// @dev allows account owner to add additional functions to the account via an override
+    fallback() external payable {
+        // _handleOverride();
+    }
+
+    /// @dev executes a low-level call against an account if the caller is authorized to make calls
+    function executeCall(
+        address to,
+        uint256 value,
+        bytes calldata data
+    ) external payable onlyAuthorized returns (bytes memory) {
+        emit TransactionExecuted(to, value, data);
+
+        _incrementNonce();
+
+        return _call(to, value, data);
+    }
+
```

Comment:
Can we add a batchExecuteCall function to make the batch call easy? 
Like this https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/samples/SimpleAccount.sol#L66